src/Drupal/Commands/generate/ApplicationFactory.php

```diff
@@ -48,10 +48,8 @@ public function discover(): array
         $global_generators = $this->discoverPsr4Generators();
 
         $module_generators = [];
-        if ($this->container->has(DrushServiceModifier::DRUSH_GENERATOR_SERVICES)) {
-            $module_generators = $this->container->get(DrushServiceModifier::DRUSH_GENERATOR_SERVICES)->getCommandList(
-            );
-        }
+        $serviceManager = \Drush\Drush::service('service.manager');
+        $module_generators = $serviceManager->getGenerators();
```

Comment:
@Chi-teck, a question about `GeneratorInfoAlter`ers: do these simply operate on a list of Symfony Console Command classes? Can I just loop over them here (once they are instantiated), and call alter on each? Seems that should work for the `alterGenerators()` method in this class, but I am not completely sure if that's always true.

I am removing the DrushDrupalKernel and not using the Symfony DI compiler pass to find and instantiate Drush commands, generators and etc. any longer, which is why the alter step is going away & needs to be replaced.

Reply:
Actually I just did a quick ad-hoc test, and discovered that these are running. I guess this is a feature of the Symfony Application object; I had presumed it was a feature of the Symfony DI container.